app/experimenter/nimbus-ui/src/components/PageResults/index.tsx

```diff
@@ -159,9 +168,19 @@ const PageResults: React.FunctionComponent<RouteComponentProps> = () => {
                     {metricHeader}
                     <AnalysisErrorAlert
                       errors={analysis.errors![metric.slug].filter(
+                        // de-dupe with findIndex and filter to selected analysis_basis + segment
                         (error, idx, self) =>
                           idx ===
-                          self.findIndex((e) => e.message === error.message),
+                            self.findIndex(
+                              (e) =>
+                                e.message === error.message &&
+                                e.segment === error.segment &&
+                                e.analysis_basis === error.analysis_basis,
+                            ) &&
+                          (error.analysis_basis === selectedAnalysisBasis ||
+                            error.analysis_basis === null) &&
+                          (error.segment === selectedSegment ||
+                            error.segment === null),
```

Comment:
If this is repeated could be handy to pull it out into a little function at the top so we can have it in one place if we need to change it later?